wallet/wallet_interface.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package wallet
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/account"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/tx"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/network"
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+type BuildTxParams struct {
+	BuildTxInput
+	Account account.Account
+	Network network.Network
+}
+
+// BuildTxInput represents a generic interface for transaction parameters
+type BuildTxInput interface {
+	// GetTxType returns the transaction type identifier
+	GetTxType() string
+	// Validate validates the parameters
+	Validate() error
+	// GetChainType returns which chain this transaction is for
+	GetChainType() string
+}
+
+type SignTxParams struct {
+	*tx.BuildTxResult
+	Account account.Account
+	Network network.Network
+}
+
+type SendTxParams struct {
+	*tx.SignTxResult
+	Account account.Account
```

Comment:
Is it possible to carry `Account` and `Network` from BuildTxParameters?

Reply:
in https://github.com/ava-labs/avalanche-tooling-sdk-go/pull/172/files#diff-17f8d1b55a277bc474edbbf7ef2dfcfeac2d6c0437e117044234a416c7313e31, account and network are now baseParams